### ISSUE TITLE
[DOC] conf.py uses old syntax for `source_suffix`, a dict value is now preferred by Sphinx

### ISSUE THREAD

**Issue by jfbu** [sympy/sympy#26893]
Probably not at all an issue, but documenting in case a problem arises in future.

During build of docs via Sphinx 8 (localized to French).
```
Conversion de  `source_suffix = '.rst'` en `source_suffix = {'.rst': 'restructuredtext'}`.
```

The `source_suffix` Sphinx configuration expects a dict and accepts a string only for backward compatibility.

cf https://github.com/sphinx-doc/sphinx/pull/12584.

I don't know if this backward compatibility is here to stay, but maybe at some point consider using the newer syntax.

**Comment by asmeurer** [comment]
Can't it just be deleted? Isn't `.rst` the default? We used to use `.txt` instead of `.rst` but that was changed a long time ago.

**Comment by jfbu** [comment]
Yes `.rst` is [the default](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-source_suffix).

Pardon now the ignorabimus question, but where is MyST input syntax used by the way?

**Comment by jfbu** [comment]
Ok, yes you can simply remove it it seems, as MyST docs says

> ``extensions = ["myst_parser"]``
> This will activate the MyST Parser extension, causing all documents with the .md extension to be parsed as MyST.

Somehow I worked recently on another project where they were using ``source_suffix`` to assign `.md` files to MyST but I learned tonight simply not seeting ``source_suffix`` at all works if you have `.rst` and `.md` files.
